install.sh

```diff
@@ -592,14 +592,20 @@ getshims() {
 
 killtree $({ set +x; } 2>/dev/null; getshims; set -x)
 
-do_unmount() {
+do_unmount_and_remove() {
     awk -v path="$1" '$2 ~ ("^" path) { print $2 }' /proc/self/mounts | sort -r | xargs -r -t -n 1 umount
+    awk -v path="$1" '$2 ~ ("^" path) { print $2 }' /proc/self/mounts | sort -r | xargs -r -t -n 1 umount rm -f 
 }
 
-do_unmount '/run/k3s'
-do_unmount '/var/lib/rancher/k3s'
-do_unmount '/var/lib/kubelet/pods'
-do_unmount '/run/netns/cni-'
+do_unmount_and_remove '/run/k3s'
+do_unmount_and_remove '/var/lib/rancher/k3s'
+do_unmount_and_remove '/var/lib/kubelet/pods'
+do_unmount_and_remove '/run/netns/cni-'
+
+# Remove CNI namespaces
+for ns in $(cd /run/netns; ls cni-*); do 
+    $SUDO ip netns del ${ns}
+done
```

Comment:
Can we do this similar to how we handle the unmounting?

```suggestion
ip netns show 2>/dev/null | grep cni- | xargs -r -t -n 1 ip netns delete
```

Reply:
1. I don't like `cd` + `ls`; why is this better than find or just actually listing the namespaces with `ip netns show`?
1. $SUDO shouldn't be used in this scope - this is a heredoc that creates the uninstall script, which doesn't have $SUDO.